imagetest/test_suites/security/image_security_test.go

```diff
@@ -0,0 +1,327 @@
+	"net.ipv4.conf.default.send_redirects":       0,
+	"net.ipv4.conf.all.rp_filter":                1,
+	"net.ipv4.conf.default.rp_filter":            1,
+	"net.ipv4.icmp_echo_ignore_broadcasts":       1,
+	"net.ipv4.icmp_ignore_bogus_error_responses": 1,
+	"net.ipv4.conf.all.log_martians":             1,
+	"net.ipv4.conf.default.log_martians":         1,
+	"kernel.randomize_va_space":                  2,
+}
+
+var specialAccountShells = map[string]string{
+	"sync":     "/bin/sync",
+	"halt":     "/sbin/halt",
+	"shutdown": "/sbin/shutdown",
+}
+
+const (
+	unattendedUpgradeConfigPath  = "/etc/apt/apt.conf.d/50unattended-upgrades"
+	unattendedUpgradeBlockDebian = `Unattended-Upgrade::Origins-Pattern`
+	unattendedUpgradeBlockUbuntu = `Unattended-Upgrade::Allowed-Origins`
+	expectedDebian               = `origin=Debian,codename=${distro_codename},label=Debian-Security`
+	expectedUbuntu               = `${distro_id}:${distro_codename}-security`
+	maxUID                       = 1000
+	maxInterval                  = 7
+	minInterval                  = 1
+	sysctlBase                   = "/proc/sys/"
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *runtest {
+		os.Exit(m.Run())
+	} else {
+		os.Exit(0)
+	}
+}
+
+// TestKernelSecuritySettings Checks that the given parameter has the given value in sysctl.
+func TestKernelSecuritySettings(t *testing.T) {
+	for key, expect := range securitySettingMap {
+		v, err := sysctlGet(key)
+		if err != nil {
+			t.Fatalf("failed getting key %s", key)
+		}
+		value, err := strconv.Atoi(v)
+		if err != nil {
+			t.Fatalf("failed convert config %s value %s to int", key, v)
+		}
+		if expect != value {
+			t.Fatalf("expected %s = %d. found %d.", key, expect, value)
+		}
+	}
+}
+
+// TestAutomaticUpdates Check automatic security updates are installed and enabled.
+func TestAutomaticUpdates(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	switch {
+	case strings.Contains(image, "rhel-8"), strings.Contains(image, "centos-8"),
+		strings.Contains(image, "centos-stream"), strings.Contains(image, "almalinux"):
+		if err := verifyServiceEnabled("dnf-automatic.timer"); err != nil {
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "rhel-7"), strings.Contains(image, "centos-7"):
+		if err := verifyServiceEnabled("yum-cron"); err != nil {
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "debian") || strings.Contains(image, "ubuntu"):
+		if err := verifyUpgradesPackageInstalled(); err != nil {
+			t.Fatal(err)
+		}
+		// Check that the security packages are marked for automatic update
+		// https://wiki.debian.org/UnattendedUpgrades
+		// https://help.ubuntu.com/community/AutomaticSecurityUpdates
+		if err := verifySecurityUpgradeEnabled(image); err != nil {
+			t.Fatal(err)
+		}
+		if err := verifyAutomaticUpdate(image); err != nil {
+			t.Fatal(err)
+		}
+	default:
+		t.Skipf("skipping test on image %s", image)
```

Comment:
i think we should fail on the default - so when we introduce a new image, it will fail (alerting us to add support) rather than skipping and appearing to pass.